python/semantic_kernel/connectors/memory/astradb/astra_client.py

```diff
@@ -0,0 +1,160 @@
+import requests
+import json
+from typing import Any, Dict, List, Optional, Union
+
+import aiohttp
+
+from semantic_kernel.connectors.memory.astradb.utils import AsyncSession
+
+class AstraClient:
+    def __init__(
+        self,
+        astra_id: str,
+        astra_region: str,
+        astra_application_token: str,
+        keyspace_name: str,
+        embedding_dim: int,
+        similarity_function: str,
+        session: Optional[aiohttp.ClientSession] = None
+    ):
+        self.astra_id = astra_id
+        self.astra_application_token = astra_application_token
+        self.astra_region = astra_region
+        self.keyspace_name = keyspace_name
+        self.embedding_dim = embedding_dim
+        self.similarity_function = similarity_function
+
+        self.request_base_url = f"https://{self.astra_id}-{self.astra_region}.apps.astra.datastax.com/api/json/v1/{self.keyspace_name}"
+        self.request_header = {
+            "x-cassandra-token": self.astra_application_token,
+            "Content-Type": "application/json",
+        }
+        self._session = session
+
+    async def _run_query(self, request_url: str, query: Dict):
+        async with AsyncSession(self._session) as session:
+            async with session.post(request_url, data=json.dumps(query), headers = self.request_header ) as response:
+                if response.status == 200:
+                    response_dict = await response.json()
+                    if "errors" in response_dict:
+                        raise Exception(f"Astra DB request error - {response_dict['errors']}")
+                    else:
+                        return response_dict
+                else:
+                    raise Exception(f"Astra DB not available. Status : {response}")
+
+    async def find_collections(self, include_detail: bool = True):
+        query = {"findCollections": {"options": {"explain": include_detail}}}
+        result = await (self._run_query(self.request_base_url, query))
```

Comment:
nit: do we need the brackets around the call?

Reply:
Remove the brackets.